c++/test/ttypes.cpp

```diff
@@ -726,8 +725,11 @@ test_named()
 
         // It should be possible to define an attribute for the named type
         Attribute attr1 = itype.createAttribute("attr1", PredType::NATIVE_UCHAR, space);
-        for (i = 0; i < ds_size[0] * ds_size[1]; i++)
-            attr_data[0][i] = (int)i; /*tricky*/
+        for (hsize_t i = 0; i < ds_size[0]; i++) {
+            for (hsize_t j = 0; j < ds_size[1]; j++) {
+                                attr_data[i][j] = (unsigned)(i * ds_size[0]] + j);
```

Comment:
Ah, sorry, I was wrong about this when I get you a "thumbs up" earlier - it should be ds_size[1] here.

Reply:
Really? OK, done.

Reply:
Yes - 'i' goes in steps of the row size (ds_size[1]) down the array.